dashboard/src/app/dashboard/last-projects/last-projects.controller.js

```diff
@@ -32,23 +34,48 @@ export class DashboardLastProjectsController {
 
     promise.then(() => {
         this.projects = cheWorkspace.getAllProjects();
+        this.updateProjectWorkspace();
         this.state = 'OK';
       },
       (error) => {
         if (error.status === 304) {
           // ok
           this.projects = cheWorkspace.getAllProjects();
+          this.updateProjectWorkspace();
           this.state = 'OK';
           return;
         }
         this.state = 'error';
       });
+
+    this.projectWorkspace = {};
   }
 
+  updateProjectWorkspace() {
+    let workspaces = this.cheWorkspace.getWorkspaces();
+    workspaces.forEach((workspace) => {
+      if (workspace.config.projects && workspace.config.projects.length > 0) {
+        let projects = workspace.config.projects;
+        projects.forEach((project) => {
+          this.projectWorkspace[project.name] = {
+            id: workspace.id,
```

Comment:
projectWorkspace[project.name]  
I guess it can be unique only within one workspace.